src/test/kotlin/no/nav/helse/flex/syketilfelle/ventetid/VentetidTest.kt

```diff
@@ -663,201 +363,1950 @@ class VentetidTest :
+
+            val melding2 =
+                skapApenSykmeldingKafkaMessage(
+                    fom = LocalDate.of(2024, Month.JULY, 17),
+                    tom = LocalDate.of(2024, Month.JULY, 17),
+                ).also { it.publiser() }
+
+            erUtenforVentetid(
+                listOf(fnr),
+                sykmeldingId = melding1.sykmelding.id,
+                ventetidRequest = VentetidRequest(),
+            ).`should be false`()
+
             erUtenforVentetid(
                 listOf(fnr),
                 sykmeldingId = melding2.sykmelding.id,
-                erUtenforVentetidRequest = ErUtenforVentetidRequest(),
+                ventetidRequest = VentetidRequest(),
+            ).`should be true`()
+
+            hentVenteperiode(
+                listOf(fnr),
+                sykmeldingId = melding2.sykmelding.id,
+                venteperiodeRequest = VenteperiodeRequest(),
+            ).venteperiode.also {
+                it!!.fom `should be equal to` LocalDate.of(2024, Month.JULY, 1)
+                it.tom `should be equal to` LocalDate.of(2024, Month.JULY, 16)
+            }
+        }
+
+        @Test
+        fun `To perioder i samme sykmelding merges likt perioder fra to sykmeldinger`() {
+            val melding =
+                skapApenSykmeldingKafkaMessage(
+                    fom = LocalDate.of(2024, Month.JULY, 1),
+                    tom = LocalDate.of(2024, Month.JULY, 16),
+                )
+
+            with(melding) {
+                val p = sykmelding.sykmeldingsperioder.first()
+                copy(
+                    sykmelding =
+                        sykmelding.copy(
+                            sykmeldingsperioder =
+                                listOf(
+                                    p.copy(),
+                                    p.copy(
+                                        fom = LocalDate.of(2024, Month.JULY, 17),
+                                        tom = LocalDate.of(2024, Month.JULY, 31),
+                                    ),
+                                ),
+                        ),
+                )
+            }.also { it.publiser() }
+
+            erUtenforVentetid(
+                listOf(fnr),
+                sykmeldingId = melding.sykmelding.id,
+                ventetidRequest = VentetidRequest(),
+            ).`should be true`()
+
+            hentVenteperiode(
+                listOf(fnr),
+                sykmeldingId = melding.sykmelding.id,
+                venteperiodeRequest = VenteperiodeRequest(),
+            ).venteperiode.also {
+                it!!.fom `should be equal to` LocalDate.of(2024, Month.JULY, 1)
+                it.tom `should be equal to` LocalDate.of(2024, Month.JULY, 16)
+            }
+        }
+    }
+
+    /**
+     * Tidligere utbetalte (perioder utenfor ventetiden) perioden teller på ventetide på neste periode hvis oppholdet
+     * mellom perioden er 16 dager eller mindre.
```

Comment:
Var litt vanskelig å lese denne. Hva med noe som "Tidligere utbetalte perioder teller på ventetiden for neste periode hvis oppholdet mellom den gamle og nye perioden er 16 dager eller mindre"?

Reply:
Ja den var litt tung. Oppdatert.